GeeksCoreLibrary/Components/OrderProcess/OrderProcess.cs

```diff
@@ -434,7 +435,43 @@ private async Task<string> HandleCheckoutModeAsync()
                         break;
                     case OrderProcessStepTypes.OrderConfirmation:
                     case OrderProcessStepTypes.OrderPending:
-                        var confirmationHtml = ReplaceEntityDataInTemplate(shoppingBasket, currentItems, step, steps, paymentMethods);
+                        var order = new WiserItemModel();
+                        string errorMessage = null;
+                        if (httpContextAccessor.HttpContext == null || !httpContextAccessor.HttpContext.Request.Query.ContainsKey("order"))
+                        {
+                            errorMessage = "Order not found, please contact us";
+                            Logger.LogError("Failed to get the order by ID during the confirmation step in the order process because query string was not set.");
+                        }
+                        else
+                        {
+                            var encryptedOrderId = httpContextAccessor.HttpContext.Request.Query["order"].ToString();
+                            if (String.IsNullOrWhiteSpace(encryptedOrderId))
+                            {
+                                errorMessage = "Order not found, please contact us";
+                                Logger.LogError("Failed to get the order by ID during the confirmation step in the order process because no order ID was given.");
+                            }
+                            else
+                            {
+                                try
+                                {
+                                    var orderId = Convert.ToUInt64(encryptedOrderId.Decrypt());
+                                    order = await wiserItemsService.GetItemDetailsAsync(orderId, entityType: Constants.OrderEntityType, skipPermissionsCheck: true);
+                                }
+                                catch (Exception exception)
+                                {
+                                    errorMessage = "Order not found, please contact us";
+                                    Logger.LogError($"Failed to get the order by ID during the confirmation step in the order process due to exception while processing encrypted ID '{encryptedOrderId}': {exception}");
+                                }
+                            }
+                        }
+
+                        var confirmationHtml = ReplaceEntityDataInTemplate(order, currentItems, step, steps, paymentMethods);
+
+                        if (!String.IsNullOrWhiteSpace(errorMessage))
```

Comment:
Ik zou de vervanging wel altijd doen, als er geen fout is, dan die variabele vervangen met lege string, anders blijft er ergens `{loadOrderError}` in de HTML staan wanneer er geen fout optreed.